eis_toolkit/spatial_analyses/cell_based_association.py

```diff
@@ -0,0 +1,511 @@
+
+    Returns:
+        Corresponding CBA object.
+    """
+
+    tmp = input_csv_file_path.split("__")
+    crs = tmp[-1].split(".")[0].replace("_", ":")
+    df = pd.read_csv(input_csv_file_path)
+    df["geometry"] = df["geometry"].apply(wkt.loads)
+    # df.insert(0, "geometry", df["geometry"].apply(wkt.loads))
+    # df["geometry"] = df["geom"].apply(wkt.loads)
+    # df = df.drop("geom", axis=1)
+    cba_grid = gpd.GeoDataFrame(df, geometry="geometry")
+    cba_grid = cba_grid.set_crs(crs)
+    cba_grid.index = df["cell_id"]
+    cba_grid = cba_grid.drop("cell_id", axis=1)
+    cba_grid = cba_grid.astype("int32", errors="ignore")
+    return cba_grid
+
+
+def from_vector_file(cba_vector_file_path: str) -> gpd.GeoDataFrame:
+    """Intermediate utility.
+
+    Read a CBA grid from a vector file (ESRI shapefile, geojson...).
+    All attributes (and all values of each attribute) are kept.
+    The geometries contained in the file are assumed to represent a regular
+    square grid.
+
+    Args:
+        cba_vector_file_path: input vector file.
+
+    Returns:
+        Corresponding CBA object.
+    """
+    geodata = gpd.GeoDataFrame.from_file(cba_vector_file_path)
+    cba_grid = gpd.GeoDataFrame(geodata, geometry="geometry")
+    cba_grid.set_index("cell_id", inplace=True)
+    cba_grid.crs = geodata.crs
+    cols = cba_grid.columns.tolist()
+    cols = cols[-1:] + cols[:-1]
+    cba_grid = cba_grid[cols]
+    cba_grid = cba_grid.astype("int32", errors="ignore")
+    return cba_grid
+
+
+def to_csv(cba: gpd.GeoDataFrame, output_path: str) -> None:
+    """Intermediate utility.
+
+    Saves the object in a CSV file.
+
+    Args:
+        output_csv_file_path: name of the saved file suffixed with the
+            CRS (spatial projection system).
+
+    Returns:
+        None
+    """
+
+    crs_txt = f"epsg:{cba.crs.to_epsg()}"
+    tmp = pd.DataFrame(cba)
+    tmp["geometry"] = tmp.geometry.apply(lambda x: wkt.dumps(x))
+    # tmp = tmp.drop("geometry", axis=1)
+    tmp["cell_id"] = cba.index
+    tmp.to_csv(output_path + "__" + str(crs_txt).replace(":", "_") + ".csv", index=False)
+
+
+def to_raster(cba: gpd.GeoDataFrame, output_path: str) -> None:
+    """Intermediate utility.
+
+    Saves the object as a raster TIFF file.
+
+    Args:
+        output_tiff_path: name of the saved file.
+
+    Returns:
+        None
+    """
+
+    crs_txt = f"EPSG:{cba.crs.to_epsg()}"
+    count = len(cba.columns.drop("geometry"))
+
+    geometries = cba["geometry"].values
+    x = np.unique(geometries.centroid.x)
+    y = np.unique(geometries.centroid.y)
+    X, Y = np.meshgrid(x, y)
+    x_resolution = X[0][1] - X[0][0]
+    y_resolution = Y[1][0] - Y[0][0]
+    min_x, min_y, max_x, max_y = cba.total_bounds
+    width = (max_x - min_x) / x_resolution
+    height = (max_y - min_y) / y_resolution
+
+    values = []
+    col_name = []
+    # for col in cba.columns.drop("geometry"):
+    for col in cba.select_dtypes(exclude=["geometry"]).columns:
```

Comment:
I believe this is incorrect usage, you should specify the dtypes rather than column names:

```suggestion
    for col in cba.select_dtypes(include=["int32", "int64"]).columns:
```

Reply:
Both ways works in my side. From what I understand, "geometry" can be used as a dtype with GeoDataFrame. I tried it with multiple geometry columns with various names and all were excluded.